### ISSUE TITLE
securecookie: the value is too long

### ISSUE THREAD

**Issue by menefotto** [auth0-samples/auth0-golang-web-app#4]
Hi the facebook login isn't working when redirected at the http://localhost:3000/callback url, it fails to save
at line 72 in /routes/callback/callback.go with the following error line securecookie: the value is too long.

**Comment by socceroos** [comment]
I can confirm that this is an issue.

The problem is that the profile retrieved from the /userinfo endpoint is too large to serialise in a cookie.

By commenting out line 65 of callback/callback.go you will be able to get a working solution.

However this means you'll need to retrieve the user's profile from Auth0 every time you need it. That's pretty terrible.

I think this issue has arisen from the Auth0 guys updating their Go web example to use gorilla sessions instead of the beego session manager. It hasn't been properly tested.

**Comment by gogo-boot** [comment]
I experience the same error with latest example code.

**Comment by icamys** [comment]
I tested this app with Keycloak and still receive this error.

**Comment by Waishnav** [comment]
> I test with Azure AD. I experience the same error with latest example code.

did you able to fix this issue?

**Comment by gogo-boot** [comment]
> > I test with Azure AD. I experience the same error with latest example code.
> 
> did you able to fix this issue?

I did, it was long time ago. I don't remember how i did exactly. I made 2 samples that time and it worked. it might help

Test with Google Oauth2
https://github.com/gogo-boot/oauth2-google

Test with Deutschebahn, they use Azure AD. (Readme is written, test with google, but it is not correct, I tested with AzureAD)
https://github.com/gogo-boot/oauth2-deutschebahn

Implementation with Refactoring
https://github.com/gogo-boot/go-boot/blob/main/platform/authenticator/auth.go
